css/css-flexbox-1/percentage-heights-001.html

```diff
@@ -0,0 +1,140 @@
+<meta name="flags" content="dom" />
+
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="support/check-layout-th.js"></script>
+
+<style>
+.rect {
+  width: 50px;
+  height: 50px;
+  background-color: green;
+}
+
+.flexbox {
+  width: 50px;
+  outline: 3px solid black;
+}
+
+
+.flexbox > * {
+  min-height: 0;
+  min-width: 0;
+}
+
+.flexbox > div > div {
+  overflow: hidden;
+}
+</style>
+
+<body onload="checkLayout('.flexbox')" style="height: 800px;">
+<div id=log></div>
+
+
+<p>This test verifies that we consider flex items' cross sizes to be definite
+if the align value is stretch (the default)</p>
+
+<p>Tests that we get a definite size in the simple case:</p>
+<div class="flexbox" data-expected-height="50">
+  <div data-expected-height="50">
+    <div style="height: 50%" data-expected-height="25">
+      <div class="rect" data-expected-height="50"></div>
+    </div>
+  </div>
+</div>
+
+<p>Tests that we get an definite size in a wrapping flexbox:</p>
+<div class="flexbox wrap" style="height: 50px;" data-expected-height="50">
+  <div data-expected-height="50">
+    <div style="height: 50%" data-expected-height="25">
+      <div class="rect" data-expected-height="50"></div>
+    </div>
+  </div>
+</div>
+
+<p>Tests that we get an indefinite size when not stretch-aligning:</p>
+<div class="flexbox wrap" style="height: 50px;" data-expected-height="50">
+  <div class="align-self-flex-start" data-expected-height="50">
+    <div style="height: 50%" data-expected-height="50">
+      <div class="rect" data-expected-height="50"></div>
+    </div>
+  </div>
+</div>
+
+<p>Tests that we get a definite size in a definite-height flexbox:</p>
+<div class="flexbox" style="height: 50px;" data-expected-height="50">
+  <div data-expected-height="50">
+    <div style="height: 50%" data-expected-height="25">
+      <div class="rect" data-expected-height="50"></div>
+    </div>
+  </div>
+</div>
+
+<p>Tests that we get an definite size in a nested flexbox where only the outer
+one has an explicit height:</p>
+<div class="flexbox" style="height: 50px;" data-expected-height="50">
+  <div class="flexbox" data-expected-height="50">
+    <div data-expected-height="50">
+      <div style="height: 50%" data-expected-height="25">
+        <div class="rect" data-expected-height="50"></div>
+      </div>
+    </div>
+  </div>
+</div>
+
+<p>Tests that we get an definite size in a nested flexbox where only the outer
+one has an explicit height and has an opposite direction:</p>
+<div class="flexbox" style="height: 50px;" data-expected-height="50">
+  <div class="flexbox column" data-expected-height="50">
+    <div class="flex-one" data-expected-height="50">
+      <div style="height: 50%" data-expected-height="25">
+        <div class="rect" data-expected-height="50"></div>
+      </div>
+    </div>
+  </div>
+</div>
+
+<p>Tests that we respect min-height:</p>
```

Comment:
Perhaps expand descriptive text to be a bit more specific, e.g.:
"Tests that we respect min-height, on child of a flex item with percent height that's treated as definite:"

ALSO: this makes me think -- it might be worth testing a *percent min/max-height* in this scenario (probably max-height since we're generally testing percent resolution by letting it produce a smaller size).   e.g. we should have something like the very first example in this file, but with "max-height: 50%" instead of "height: 50%" on the child of the flex item.  That should end up with a height of 25, I think.